bluetoe/link_layer/include/bluetoe/peripheral_latency.hpp

```diff
@@ -0,0 +1,236 @@
+
+        /**
+         * @brief listen, if the link layer contains unacknowledged data.
+         *
+         * If a PDU was send out, but no acknowledgement was jet received,
+         * the link layer will listen at the next connection event, if
+         * this options is set.
+         *
+         * This might be interesting, if the sending queue is very small because
+         * Bluetoe has to keep a copy of a transmitted PDU, as long as that PDU
+         * is not acknowledged.
+         */
+        listen_if_unacknowledged_data,
+
+        /**
+         * @brief listen to the next connection event, if the last received PDU was
+         *        not empty.
+         */
+        listen_if_last_received_not_empty,
+
+        /**
+         * @brief listen to the next connection event, if the last transmitted PDU was
+         *        not empty.
+         */
+        listen_if_last_transmitted_not_empty,
+
+        /**
+         * @brief listen to the next connection event, if the last received PDU had
+         *        the more data (MD) flag set.
+         *
+         * With the MD flag, the central (and the peripheral) can indicate, that there
+         * is more data to be send. Usually this data is then transmitted during a
+         * connection even. If there is no support for the MD flag either by the peripheral
+         * or by the central, then, this option might be usefull.
+         */
+        listen_if_last_received_had_more_data,
+
+        /**
+         * @brief listen to the next connection event, if for the last connection event,
+         *        a crc error happend, or a timeout occured.
+         *
+         * This behavior is required by the Bluetooth specification and should not left out
+         * without good reason.
+         */
+        listen_if_last_received_had_error,
+
+    };
+
+    /**
+     * @brief defines a peripheral configuration to be used by the link layer
+     *
+     * The configuration is a set of options from peripheral_latency. Example:
+     *
+     * @code{.cpp}
+     * bluetoe::device<
+     *    gatt_server_definition,
+     *    bluetoe::link_layer::peripheral_latency_configuration<
+     *        bluetoe::link_layer::peripheral_latency::listen_if_unacknowledged_data,
+     *        bluetoe::link_layer::peripheral_latency::listen_if_last_received_had_error
+     *    >
+     * > gatt_server;
+     * @endcode
+     *
+     * @sa peripheral_latency
+     * @sa peripheral_latency_configuration_set
+     */
+    template < peripheral_latency ... Options >
+    struct peripheral_latency_configuration
+    {
+        /** @cond HIDDEN_SYMBOLS */
+        struct meta_type :
+            details::peripheral_latency_meta_type,
+            details::valid_link_layer_option_meta_type {};
+        /** @endcond */
+    };
+
+    /**
+     * @brief allows to change the peripheral latency configuration to be changed at runtime
+     *        between a given set of configurations.
```

Comment:
So @TorstenRobitzki do you think this commit will directly add the support to change this at runtime?

Reply:
Support to change configuration at runtime would be configured using the `peripheral_latency_configuration_set<>` configuration option.